### PR TITLE
Fix some Riivolution file patching edge cases.

### PR DESCRIPTION
**Pull request by AdmiralCurtiss**
This fixes the booting of Wii Sports Resort Storm Island, [as reported on the forums](https://forums.dolphin-emu.org/Thread-wii-sports-resort-storm-island-freezes-at-intro-screen-riivolution).

### CODE REVIEW

File: Source/Core/DiscIO/RiivolutionPatcher.cpp [hunk 1]
```diff
@@ -364,148 +364,109 @@ static FSTBuilderNode* FindFileNodeInFST(std::string_view path, std::vector<FSTB
...
+    const auto combine_paths = [](std::string_view a, std::string_view b) {
+      if (a.empty())
+        return std::string(b);
+      if (b.empty())
+        return std::string(a);
+      if (StringEndsWith(a, "/"))
+        a.remove_suffix(1);
+      if (StringBeginsWith(b, "/"))
+        b.remove_prefix(1);
+      return std::string(a) + "/" + std::string(b);
```
Comment (iwubcode) [inline Source/Core/DiscIO/RiivolutionPatcher.cpp:430]: NBD but I think `fmt::format("{}/{}", a, b)` may be a bit more performant

**Comment by AdmiralCurtiss** [comment]
Hold on, this apparently breaks Newer...

File: Source/Core/DiscIO/RiivolutionPatcher.cpp (hunk 1 above)
Comment (AdmiralCurtiss) [reply Source/Core/DiscIO/RiivolutionPatcher.cpp:430]: Good point, yes.

File: Source/Core/DiscIO/RiivolutionPatcher.cpp [hunk 2]
```diff
@@ -364,148 +364,109 @@ static FSTBuilderNode* FindFileNodeInFST(std::string_view path, std::vector<FSTB
                            create_if_not_exists);
 }
 
-static void FindFilenameNodesInFST(std::vector<DiscIO::FSTBuilderNode*>* nodes_out,
-                                   std::string_view filename, std::vector<FSTBuilderNode>* fst)
+static DiscIO::FSTBuilderNode* FindFilenameNodeInFST(std::string_view filename,
+                                                     std::vector<FSTBuilderNode>& fst)
 {
-  for (FSTBuilderNode& node : *fst)
+  for (FSTBuilderNode& node : fst)
```
Comment (iwubcode) [inline Source/Core/DiscIO/RiivolutionPatcher.cpp:370]: Are you able to make this `const` ?

Comment (AdmiralCurtiss) [reply Source/Core/DiscIO/RiivolutionPatcher.cpp:370]: No, because this is potentially the return value and thus must be mutable.

**Comment by AdmiralCurtiss** [comment]
I genuinely have no idea why FifoCI seems to think this affects rendering. Help?

File: Source/Core/DiscIO/RiivolutionPatcher.cpp (hunk 2 above)
Comment (iwubcode) [reply Source/Core/DiscIO/RiivolutionPatcher.cpp:372]: Yeah, it looks like all the usages were just getting data on the item but I wasn't 100% sure.  NBD either way

**Review by leoetlino — APPROVED** [review]

**Pull request merged**